app/models/search_log.rb

```diff
@@ -103,19 +103,38 @@ def self.term_details(term, period = :weekly, search_type = :all)
   end
 
   def self.trending(period = :all, search_type = :all)
-    result = SearchLog.select("term,
-       COUNT(*) AS searches,
-       SUM(CASE
+    SearchLog.trending_from(start_of(period), search_type: search_type)
+  end
+
+  def self.trending_from(start_date, options = {})
```

Comment:
I think this new method needs to be tested.

Reply:
I'm mixed on this, as it's already tested by this existing one and is adding more code to add very little confidence. But ok fine, won’t fight for this.

Reply:
My worry is that the existing tests does not cover all the other options that the method accepts.